juju/juju
# Substitute model-uuid in placement with actual uuid in DeployFromRepository

We replace the placement scope `model-uuid` with the actual model-uuid before making the Deploy call to the api. This substitution code was somewhat hidden in the old deploy code, so it's missed when we switched to deploying on the server side (i.e. the DeployFromRepository).

Fixes https://bugs.launchpad.net/juju/+bug/2056501

## Checklist




## QA steps

Setup a MAAS ([see discourse post for it](https://discourse.charmhub.io/t/local-maas-development-setup/1319)).

Add a node:
```
 $ juju add-machine node-1 --base ubuntu@22.04
```

Deploy on that node:
```
 $ juju deploy ubuntu --base ubuntu@22.04 --to node-1
```

Without this change, you'll see:
```
ERROR cannot add application "ubuntu": placement scope: invalid model UUID "model-uuid"
ERROR failed to deploy charm "ubuntu"
```

With this change, the deploy should succeed with a message like the following:
```
Deployed "ubuntu-base" from charm-hub charm "ubuntu", revision 24 in channel stable on ubuntu@22.04/stable
```

## Links



**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2056501

**Jira card:** JUJU-
